src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -174,10 +174,40 @@ private function createRelationsForItem($item, $formattedRelations)
                     $relation->dissociate()->save();
                 }
             } elseif ($relation instanceof HasOne || $relation instanceof MorphOne) {
-                $modelInstance = $relation->updateOrCreate([], $relationDetails['values']);
+                // For HasOne and MorphOne relationships, the dev might want to a few different things:
+                // (A) save an attribute on the related entry (eg. passport.number)
+                // (B) save an entire related entry (eg. passport)
+                // (C) delete the entire related entry (eg. passport)
+                // so let's see which one it is, from $relationDetails, which will contain:
+                // [
+                //     'model' => 'App\Models\Passport',
+                //     'parent' => 'App\Models\Pet',
+                //     'values' => **THE TRICKY BIT**,
+                // ]
+                // And the values can be:
+                //      - (A) ['number' => 1315, 'name' => 'Something'] (if passed using a text/number/etc field)
+                //      - (B) ['passport' => [['number' => 1314, 'name' => 'Something']]] (if passed using a repeatable field)
+                //      - (C) ['passport' => null] (if deleted from the repeatable field)
+                $relation_values = $relationDetails['values'][$relationMethod] ?? $relationDetails['values'] ?? null;
```

Comment:
@pxpm I've added a few comments (actually a few _too many_ comments) to this bit to understand it, and to be able to highlight the most common scenario. And in the process, I think I've found a hole we need to plug somehow.

What about this scenario?

```
DB Tables:
- user: id, name
- slug: id, user_id, slug

Relationships:
- User hasOne Slug
- Slug belongsTo User
```

Notice that `slug` is the table name, the model name BUT ALSO the name of the attribute on `slug`. In this rare (but possible) scenario, the functionality stops working:
- with `CRUD::field('slug.slug');` the saving will error at `if (count($relation_values) != count($relation_values, COUNT_RECURSIVE)) {`
- with `CRUD::field('slug')->fields([ ['name' => 'slug', 'type' => 'text'], ]);` the saving will not error, but will not work either (nothing gets saved)

I think we need to account for this edge case. The _problem_ I'm identifying is HOW we judge that what the fields wants is to do A, B or C - we check `$relationDetails['values'][$relationMethod]` - but in this case `$relationDetails['values']['slug']` isn't explicit enough in what it wants us to do, it can be:
- `string` or `null` for Scenario A
- `array of arrays` for Scenario B
- `null` for Scenario C (aka removing)

I've tinkered with a solution for a few hours and I'll submit a PR towards this branch, please take a look and double-check my work.

Reply:
See https://github.com/Laravel-Backpack/CRUD/pull/4075 please, and merge it to this branch and tell me if it's all ok.

Reply:
Also, could you please add unit tests for all of these scenarios? I think it'll save us some trouble in the future.